include/oneapi/dpl/pstl/hetero/algorithm_ranges_impl_hetero.h

```diff
@@ -564,6 +568,111 @@ __pattern_minmax_element(_ExecutionPolicy&& __exec, _Range&& __rng, _Compare __c
+    auto __tmp_out_values =
+        oneapi::dpl::__par_backend_hetero::__internal::__buffer<_ExecutionPolicy, __val_type>(__exec, __n).get_buffer();
+
+    //create two views over keys, the first with the first key removed for adjacent key comparison
+    auto __k1 = oneapi::dpl::__ranges::drop_view_simple<_Range1, __diff_type>(__keys, 1);
+    auto __k2 = __keys;
+
+    // view1 elements are a tuple of the element index and pairs of adjacent keys
+    // view2 elements are a tuple of the elements where key-index pairs will be written by copy_if
+    auto __view1 = experimental::ranges::zip_view(experimental::ranges::views::iota(0, __n), __k1, __k2);
+    auto __view2 = experimental::ranges::zip_view(experimental::ranges::views::all_write(__tmp_out_keys),
+                                                  experimental::ranges::views::all_write(__idx));
+
+    // use workgroup size as the maximum segment size.
+    ::std::size_t __wgroup_size = oneapi::dpl::__internal::__max_work_group_size(__exec);
+    // change __wgroup_size according to local memory limit
+    __wgroup_size = oneapi::dpl::__internal::__max_local_allocation_size(
+        ::std::forward<_ExecutionPolicy>(__exec), sizeof(__key_type) + sizeof(__val_type), __wgroup_size);
+
+    // element is copied if it is the last element (marks end of final segment), is in an index
+    // evenly divisible by wg size (ensures segments are not long), or has a key not equal to the
+    // adjacenent element (marks end of real segments)
+    auto __res = __pattern_copy_if(::std::forward<_ExecutionPolicy>(__exec), __view1, __view2,
+                                   [__n, __binary_pred, __wgroup_size](const auto& __a) {
+                                       return ::std::get<0>(__a) == __n ||
+                                              ::std::get<0>(__a) % __wgroup_size == 0 ||              // segment size
+                                              !__binary_pred(::std::get<1>(__a), ::std::get<2>(__a)); //keys comparison
+                                   },
+                                   unseq_backend::__brick_assign{});
+
+    //reduce by segment
+    oneapi::dpl::__par_backend_hetero::__parallel_for(
+        ::std::forward<_ExecutionPolicy>(__exec), unseq_backend::__brick_reduce_idx<_BinaryOperator>{__binary_op},
+        __res, experimental::ranges::views::all_read(__idx), ::std::forward<_Range2>(__values),
+        experimental::ranges::views::all_write(__tmp_out_values))
+        .wait();
+
+    // Round 2: final reduction to get result for each segment of equal adjacent keys
+
+    // create views over adjacent keys
+    oneapi::dpl::__ranges::all_view<__key_type, __par_backend_hetero::access_mode::read_write> __new_keys(
+        __tmp_out_keys);
+    auto __k3 = oneapi::dpl::__ranges::drop_view_simple<
+        oneapi::dpl::__ranges::all_view<__key_type, __par_backend_hetero::access_mode::read_write>, __diff_type>(
+        __new_keys, 1);
+    auto __k4 = __new_keys;
```

Comment:
`__k4` is also used only once, so seems redundant.

Reply:
Removed.